hudi-sync/hudi-sync-common/src/main/java/org/apache/hudi/sync/common/HoodieSyncClient.java

```diff
@@ -121,10 +121,25 @@ public MessageType getStorageSchema(boolean includeMetadataField) {
     try {
       return tableSchemaResolver.getTableParquetSchema(includeMetadataField);
     } catch (Exception e) {
-      throw new HoodieSyncException("Failed to read schema from storage.", e);
+      throw new HoodieSyncException(buildSchemaReadErrorMessage(e), e);
     }
   }
 
+  private String buildSchemaReadErrorMessage(Exception e) {
+    String errorMessage = e.getMessage() != null ? e.getMessage() : e.getClass().getName();
+    if (e instanceof java.io.FileNotFoundException) {
+      return String.format(
+          "Cannot read Hudi table schema.%n%n"
```

Comment:
If we hit FileNotFoundException, then it means, the data file was deleted by some means right. 

1. Aggressive cleaner retention compared to query run times
2. Manual file deletions (timeline files or data files)
3. Concurrent writers without proper locking or configurations set

Reply:
hey @yihua : this is specifically when FileNotFoundException. not sure if the error msg lines up with that.